cmd/influxd/run/server.go

```diff
@@ -508,6 +513,41 @@ func (s *Server) reportServer() {
 	go client.Post("http://m.influxdb.com:8086/db/reporting/series?u=reporter&p=influxdb", "application/json", data)
 }
 
+// registerServer registers the server on start-up.
+func (s *Server) registerServer() error {
+	if s.enterpriseToken == "" {
+		return nil
+	}
+
+	clusterID, err := s.MetaStore.ClusterID()
+	if err != nil {
+		log.Printf("failed to retrieve cluster ID for registration: %s", err.Error())
+		return err
+	}
+
+	hostname, err := os.Hostname()
+	if err != nil {
+		return err
+	}
+
+	j := map[string]interface{}{
+		"cluster_id": clusterID,
+		"server_id":  s.MetaStore.NodeID(),
+		"host":       hostname,
+		"product":    "influxdb",
+		"version":    s.buildInfo.Version,
+	}
+	b, err := json.Marshal(j)
+	if err != nil {
+		return err
+	}
+
+	url := fmt.Sprintf("%s/api/v1/servers?token=%s", s.enterpriseURL, s.enterpriseToken)
+	client := http.Client{Timeout: time.Duration(5 * time.Second)}
+	go client.Post(url, "application/json", bytes.NewBuffer(b))
```

Comment:
Yeah, that might be best since the upper layers are not going to do
anything differently even if there is an error.

On Thu, Oct 15, 2015 at 1:18 PM, Mark Bates <notifications@github.com>
wrote:

> In cmd/influxd/run/server.go
> <https://github.com/influxdb/influxdb/pull/4459#discussion_r42174608>:
>
> > +
> > +	j := map[string]interface{}{
> > +		"cluster_id": clusterID,
> > +		"server_id":  s.MetaStore.NodeID(),
> > +		"host":       hostname,
> > +		"product":    "influxdb",
> > +		"version":    s.buildInfo.Version,
> > +	}
> > +	b, err := json.Marshal(j)
> > +	if err != nil {
> > +		return err
> > +	}
> > +
> > +	url := fmt.Sprintf("%s/api/v1/servers?token=%s", s.enterpriseURL, s.enterpriseToken)
> > +	client := http.Client{Timeout: time.Duration(5 * time.Second)}
> > +	go client.Post(url, "application/json", bytes.NewBuffer(b))
>
> You could still call it async and log errors:
>
> go func() {
>   client.Post(url, "application/json", bytes.NewBuffer(b))
>   // error logging here.
> }
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/influxdb/influxdb/pull/4459/files#r42174608>.
>